examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+import type { DataSource, syncCollection } from "./data"
+
+import { ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { useState, useMemo, useLayoutEffect } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields } from "./data"
+import { Spinner } from "./components/Spinner"
+import { UI_DEFAULTS } from "./constants"
+
+interface FieldMappingRowProps {
+    originalField: ManagedCollectionField
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+function FieldMappingRow({ originalField, field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) {
+    const isUnsupported = !field
+    const hasFieldNameChanged = field!.name !== originalField.name
+    const fieldName = hasFieldNameChanged ? field!.name : ""
+    const placeholder = isUnsupported ? "Unsupported Field" : originalField.name
+    const isDisabled = isUnsupported || isIgnored
+
+    return (
+        <>
+            <div
+                className="source-field"
+                aria-disabled={isDisabled}
+                onClick={() => onFieldToggle(field!.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    disabled={isUnsupported}
+                    checked={!isDisabled}
+                    onChange={event => {
+                        event.stopPropagation()
+                    }}
+                />
+                <span>{originalField.name}</span>
+            </div>
+            <svg xmlns="http://www.w3.org/2000/svg" width="8" height="16">
+                <path
+                    d="M 3 11 L 6 8 L 3 5"
+                    fill="transparent"
+                    strokeWidth="1.5"
+                    stroke="#999"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                ></path>
+            </svg>
```

Comment:
What is this icon? Let's turn it into a component as well

Reply:
This is the chevron icon! I moved it into a component.

<img width="23" alt="image" src="https://github.com/user-attachments/assets/bf08d495-23f8-4652-ad22-7ce83d795c63" />